pkg/jobs/update.go

```diff
@@ -343,6 +349,86 @@ func (ju *JobUpdater) UpdateHighwaterProgressed(highWater hlc.Timestamp, md JobM
 	return nil
 }
 
+func (ju *JobUpdater) PauseRequested(
+	ctx context.Context, txn isql.Txn, md JobMetadata, reason string,
+) error {
+	fn, err := ju.PauseRequestFunc(md.Payload)
+	if err != nil {
+		return err
+	}
+	return ju.PauseRequestedWithFunc(ctx, txn, md, fn, reason)
+}
+
+func (ju *JobUpdater) PauseRequestedWithFunc(
+	ctx context.Context, txn isql.Txn, md JobMetadata, fn onPauseRequestFunc, reason string,
+) error {
+	if md.Status == StatusPauseRequested || md.Status == StatusPaused {
+		return nil
+	}
+	if md.Status != StatusPending && md.Status != StatusRunning && md.Status != StatusReverting {
+		return fmt.Errorf("job with status %s cannot be requested to be paused", md.Status)
+	}
+	if fn != nil {
+		if err := ju.PauseRequestFuncRunner(ctx, fn, md); err != nil {
```

Comment:
according to the `jobUpdater` doctstring, it is merely responsible for accumulating metadata changes to be persisted  to disk, but with this change, the `jobUpdater` now executes the `onPauseRequestFunc`, which I think can do much more than update in memory metadata.

 This is making it harder for me to distinguish what an `updater` does and what a `jobUpdater` does. Could you add some comments to `updater` which explains what it does and its relation to `jobUpdater`? Further, could you add some commentary to `jobUpdater` definition that explains what `jobUpdater.PauseRequestFunc` and `jobUpdater.PauseRequestFuncRunner` do?

Reply:
Hrm. Now i feel like we should just delete this `updater/JobUpdater.PauseRequestedWithFunc` logic. The only user of it is changefeeds, which use it to update their job status. I think they can update their job status in a much less complicated way. Further, no one uses the `PauseRequesterInterface`.

Reply:
we could also, land this refactoring, and chat with cdc about removing `PauseRequestedWithFunc` after the fact.

Reply:
Thanks for taking a second look at this.  Given that as far as I can tell no jobs _actually_ implement OnPauseRequest, we can completely remove that method. That allows us to simplify the implementation of PauseRequestedWithFunc such that the caller just gets passed the job updater which I think will fix the semantic problems.

Reply:
Let's do that in this PR or we will never do it.